src/UI/IdentityBuilderUIExtensions.cs

```diff
@@ -36,6 +36,11 @@ public static IdentityBuilder AddDefaultUI(this IdentityBuilder builder)
                     .MakeGenericType(builder.UserType));
             builder.Services.TryAddTransient<IEmailSender, EmailSender>();
 
+            if (builder.UserType == typeof(IdentityUser))
+            {
+                builder.Services.TryAddSingleton<IUIUserInstanceFactory<IdentityUser>, DefaultUIUserInstanceFactory>();
```

Comment:
This will not work, you need to register it as TUser, if you get a custom user this will not be registered and we don't want people to have to register a custom factory for creating users. This will not even work in the default templates as they use ApplicationUser

Reply:
Yeah I commented earlier, maybe on an older version, but this should be smarter and user reflection and set things up properly for any derived `IdentityUser<TKey>` derived `TUser` type